File: src/cleansia_android/customer-app/src/main/java/cz/cleansia/customer/features/orders/WorkContractAcceptanceLine.kt
package cz.cleansia.customer.features.orders

import cz.cleansia.customer.core.orders.OrderDetailDto

/** One crew member's acceptance of the contract for work, as the customer's detail states it. */
data class WorkContractAcceptanceLine(
    val id: String,
    /** The crew entry's name as the server masked it; null when it sent none. */
    val cleanerName: String?,
    /** ISO-8601 date-time. */
    val acceptedOn: String,
    val documentVersion: String,
)

/**
 * The acceptance carries no name; the crew entry whose id is the acceptance's seat does, already
 * masked for the customer's eyes. Pairing them here keeps one masking path. An acceptance naming no
 * current seat, or missing what the line reads, is dropped — a line that cannot say who or when says
 * nothing the crew card does not.
 */
fun OrderDetailDto.workContractAcceptanceLines(): List<WorkContractAcceptanceLine> {
    val crew = assignedEmployees.orEmpty()
    return workContractAcceptances.orEmpty().mapNotNull { acceptance ->
        val seat = crew.firstOrNull { it.id != null && it.id == acceptance.orderEmployeeId }
            ?: return@mapNotNull null
        WorkContractAcceptanceLine(
            id = acceptance.id ?: return@mapNotNull null,
            cleanerName = seat.fullName?.takeIf { it.isNotBlank() },
            acceptedOn = acceptance.acceptedOn ?: return@mapNotNull null,
            documentVersion = acceptance.documentVersion.orEmpty(),
        )
    }
}

File: src/cleansia_ios/CleansiaCustomer/Sources/Features/Orders/WorkContractAcceptanceLine.swift
import Foundation

/// One crew member's acceptance of the contract for work, as the customer's detail states it.
struct WorkContractAcceptanceLine: Equatable, Identifiable {
    let id: String
    /// The crew entry's name as the server masked it; nil when it sent none.
    let cleanerName: String?
    let acceptedOn: Date
    let documentVersion: String
}

extension CustomerOrderDetail {
    /// The acceptance carries no name; the crew entry whose id is the acceptance's seat does, already
    /// masked for the customer's eyes. Pairing them here keeps one masking path. An acceptance naming
    /// no current seat is dropped — a line that cannot say who says nothing the crew card does not.
    func workContractAcceptanceLines() -> [WorkContractAcceptanceLine] {
        workContractAcceptances.compactMap { acceptance in
            guard let seat = assignedEmployees.first(where: { $0.id == acceptance.orderEmployeeId }) else {
                return nil
            }
            return WorkContractAcceptanceLine(
                id: acceptance.id,
                cleanerName: seat.fullName.flatMap { $0.isBlank ? nil : $0 },
                acceptedOn: acceptance.acceptedOn,
                documentVersion: acceptance.documentVersion
            )
        }
    }
}
